en/docs/develop/writing-a-custom-federated-authenticator.md

```diff
@@ -106,1262 +111,118 @@ public interface ApplicationAuthenticator extends Serializable {
-        ctxt.getBundleContext().registerService(ApplicationAuthenticator.class.getName(),
-            facebookAuthenticator, props);
-        if (LOGGER.isDebugEnabled()) {
-            LOGGER.debug("Facebook Custome Authenticator bundle is activated");
-        }
-        } catch (Throwable e) {
-        LOGGER.fatal(" Error while activating Facebook authenticator ", e);
-        }
-        }
-        protected void deactivate(ComponentContext ctxt) {
-        if (LOGGER.isDebugEnabled()) {
-        LOGGER.debug("Facebook Custom Authenticator bundle is deactivated");
-        }
-        }
-        }
-        ```
+1. Start the server and log in to the WSO2 IS Management Console.
 
-### Custom authenticator for Google
+2. Then go to the Service Providers under the main tab. Then add Service Provider name and register it. Let’s use the 
+   playground app and refer [this](https://is.docs.wso2.com/en/latest/learn/deploying-the-sample-app/#deploying-the-playground2-webapp) 
+   to configure playground app.
 
-This section includes the code used to write the custom authenticator
-for Google.
+3. Then List the Service Providers and edit the service provider as follows by navigation to the** OAuth/OpenID Configuration** under **Inbound Authentication Configuration **as explained above.
 
-1.  Download the source code from
-    [here](https://svn.wso2.org/repos/wso2/people/thanuja/org.wso2.carbon.identity.application.authenticator.social/)
-    using the following command on your terminal.
+4. Then click Configure and add[ http://localhost:8080/playground2/oauth2client](http://localhost:8080/playground2/oauth2client) as the call back URL. Click Update.
 
-    ``` java
-    $ svn checkout https://svn.wso2.org/repos/wso2/people/thanuja/org.wso2.carbon.identity.application.authenticator.social/
-    ```
+5. Navigate to **Local & Outbound Authentication Configuration** as follows, and you can find Authentication Type. 
+   Select **Federated Authentication** and select the configured federated authenticator and update to save the changed 
+   configurations.
 
-2.  Navigate to the folder you just downloaded, which contains the
-    `           pom.xml          ` file and build the source code by
-    running the following command on your terminal.
+![Partner identity provider](../assets/img/using-wso2-identity-server/partner-identity-provider.png)
 
-    ``` java
-    $ mvn clean install
-    ```
+### Try the scenario
 
-3.  Copy the
-    `          org.wso2.carbon.identity.application.authenticator.custom.google­5.0.0.jar         `
-    file found inside the **target** folder and paste it in the
-    `          <IS_HOME>/repository/components/dropins         ` folder.
+1. Access the playground app by using[ http://localhost:8080/playground2](http://localhost:8080/playground2). 
+   
+2. Then it will redirect to the login page of the application which is configured in the partner identity server which acts as the external service. 
```

Comment:
```suggestion
2. This will redirect to the login page of the application which is configured in the partner identity server which acts as the external service. 
```

Reply:
Fixed https://github.com/wso2/docs-is/pull/2527/commits/ffc80dd6ac9088892baa7c56e5f03a9aa0768d3f